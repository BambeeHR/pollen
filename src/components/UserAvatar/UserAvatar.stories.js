import { boolean, text } from '@storybook/addon-knobs';
import UserAvatar from './UserAvatar.vue';

export default {
  title: 'UserAvatar',
  component: UserAvatar,
  parameters: {
    componentSubtitle: 'A user avatar image and name component.',
  },
};

const image = 'https://cdn.filestackcontent.com/enCa5BTVeIsLHcgkdpAc';

export const Default = () => ({
  components: { UserAvatar },
  data() {
    return { image };
  },
  template:
    '<UserAvatar :image="image" name="Xavier Oaxaca" title="HR Manager" />',
});

export const Vertical = () => ({
  components: { UserAvatar },
  data() {
    return { image };
  },
  template:
    '<UserAvatar :image="image" name="Xavier Oaxaca" title="HR Manager" vertical />',
});

export const NoImage = () => ({
  components: { UserAvatar },
  template: '<UserAvatar name="Xavier Oaxaca" title="HR Manager" />',
});

export const NoNameOrImage = () => ({
  components: { UserAvatar },
  template: '<UserAvatar title="HR Manager" />',
});

export const WithKnobs = () => {
  return {
    components: { UserAvatar },
    props: {
      image: {
        default: text('Image', image),
      },
      name: {
        default: text('Name', 'Xavier Oaxaca'),
      },
      title: {
        default: text('Title', 'HR Manager'),
      },
      vertical: {
        default: boolean('Vertical', false),
      },
    },
    template: `
      <UserAvatar
        :image="image"
        :name="name"
        :title="title"
        :vertical="vertical"
      />
    `,
  };
};

/* eslint-disable no-param-reassign */
[Default, Vertical, NoImage, NoNameOrImage, WithKnobs].forEach((item) => {
  item.story = {
    parameters: {
      jest: ['BaseButton.test.js'],
    },
  };
});
