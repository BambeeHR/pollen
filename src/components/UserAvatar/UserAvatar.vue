<template>
  <div
    class="user-avatar"
    :class="[
      `user-avatar--${size}`,
      {
        'user-avatar--vertical': vertical,
        'user-avatar--horizontal': !vertical,
      },
    ]"
  >
    <AvatarImage :image="image" :initials="initials" :size="avatarSize" />
    <div v-if="name || title" class="user-avatar__details">
      <div v-if="name" class="user-avatar__name">{{ name }}</div>
      <div v-if="title" class="user-avatar__title">{{ title }}</div>
    </div>
  </div>
</template>

<script>
import AvatarImage from '../AvatarImage/AvatarImage.vue';
import getNameInitials from '../../util/getNameInitials';

export const Sizes = Object.freeze({
  TINY: 'tiny',
  SMALL: 'small',
  MEDIUM: 'medium',
  LARGE: 'large',
});

/**
 * A component for displaying a user profile photo next to their name and job
 * title.
 */
export default {
  components: { AvatarImage },
  props: {
    /** A url to the user's photo. */
    image: {
      type: String,
      default: null,
    },
    /** The user's name. */
    name: {
      type: String,
      default: null,
    },
    /** The size of the component. One of `small`, `medium`, and `large`. */
    size: {
      type: String,
      default: Sizes.MEDIUM,
      validator: (value) => Object.values(Sizes).includes(value),
    },
    /** The user's job title. */
    title: {
      type: String,
      default: null,
    },
    /** If true, displays the user name and title below their image, centered. */
    vertical: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    avatarSize() {
      switch (this.size) {
        case Sizes.TINY:
          return this.vertical ? 48 : 16;
        case Sizes.SMALL:
          return this.vertical ? 48 : 24;
        case Sizes.LARGE:
          return this.vertical ? 104 : 40;
        case Sizes.MEDIUM:
        default:
          return this.vertical ? 64 : 32;
      }
    },
    initials() {
      return getNameInitials(this.name);
    },
  },
};
</script>

<style scoped>
.user-avatar {
  @apply inline-flex;
}

.avatar-image {
  @apply flex-shrink-0;
}

.user-avatar__details {
  @apply flex flex-col flex-grow justify-center;
}

.user-avatar--horizontal .user-avatar__details {
  @apply ml-3;
}

.user-avatar__name {
  @apply font-body-small font-bold text-black;
}

.user-avatar--small .user-avatar__name {
  @apply font-body-caption;
}

.user-avatar__title {
  @apply font-body-caption text-gray-3;
}

.user-avatar__name + .user-avatar__title {
  margin-top: 0.125em;
}

.user-avatar--vertical {
  @apply flex-col items-center text-center;
}
.user-avatar--vertical .user-avatar__details {
  @apply mt-4;
}

.user-avatar--vertical .user-avatar__name + .user-avatar__title {
  @apply mt-1;
}

.user-avatar--vertical.user-avatar--tiny .user-avatar__name,
.user-avatar--vertical.user-avatar--small .user-avatar__name {
  @apply font-body-text;
}

.user-avatar--vertical.user-avatar--medium .user-avatar__name,
.user-avatar--vertical.user-avatar--large .user-avatar__title {
  @apply font-body-medium;
}

.user-avatar--vertical.user-avatar--medium .user-avatar__title {
  @apply font-body-small;
}

.user-avatar--vertical.user-avatar--large .user-avatar__name {
  @apply font-heading-title;
}

.user-avatar--tiny.user-avatar--horizontal .user-avatar__name,
.user-avatar--tiny.user-avatar--horizontal .user-avatar__title {
  @apply font-body-legal;
}
</style>
