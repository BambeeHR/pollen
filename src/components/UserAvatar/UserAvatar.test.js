import { mount } from '@vue/test-utils';
import UserAvatar from './UserAvatar.vue';

describe('UserAvatar', () => {
  test('renders correctly', () => {
    const wrapper = mount(UserAvatar, {
      propsData: {
        image: 'https://cdn.filestackcontent.com/enCa5BTVeIsLHcgkdpAc',
        name: 'Xavier Oaxaca',
        title: 'HR Manager',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('displays intials if there is no image', () => {
    const wrapper = mount(UserAvatar, {
      propsData: {
        name: 'Xavier Oaxaca',
        title: 'HR Manager',
      },
    });
    expect(wrapper.text()).toContain('XO');
  });
});
