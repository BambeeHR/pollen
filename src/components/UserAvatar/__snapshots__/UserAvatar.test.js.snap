// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar renders correctly 1`] = `
<div
  class="user-avatar user-avatar--medium user-avatar--horizontal"
>
  <div
    class="avatar-image avatar-image--rounded"
    style="background-image: url(https://cdn.filestackcontent.com/resize=width:32,height:32/enCa5BTVeIsLHcgkdpAc); font-size: 0.875rem; height: 2rem; width: 2rem;"
  >
    <!---->
  </div>
   
  <div
    class="user-avatar__details"
  >
    <div
      class="user-avatar__name"
    >
      Xavier Oaxaca
    </div>
     
    <div
      class="user-avatar__title"
    >
      HR Manager
    </div>
  </div>
</div>
`;

exports[`UserAvatar renders gallery correctly 1`] = `
<div
  class="flex"
>
  <div
    class="mr-12"
  >
    <h1
      class="type-overline type-overline-large"
    >
      
        horizontal
      
    </h1>
     
    <div
      class="my-4"
    >
      <h2
        class="type-overline type-overline-small"
      >
        tiny
      </h2>
       
      <div
        class="user-avatar user-avatar--tiny user-avatar--horizontal"
      >
        <div
          class="avatar-image avatar-image--rounded"
          style="background-image: url(https://cdn.filestackcontent.com/resize=width:16,height:16/enCa5BTVeIsLHcgkdpAc); font-size: 0.4375rem; height: 1rem; width: 1rem;"
        >
          <!---->
        </div>
         
        <div
          class="user-avatar__details"
        >
          <div
            class="user-avatar__name"
          >
            Xavier Oaxaca
          </div>
           
          <div
            class="user-avatar__title"
          >
            HR Manager
          </div>
        </div>
      </div>
    </div>
    <div
      class="my-4"
    >
      <h2
        class="type-overline type-overline-small"
      >
        small
      </h2>
       
      <div
        class="user-avatar user-avatar--small user-avatar--horizontal"
      >
        <div
          class="avatar-image avatar-image--rounded"
          style="background-image: url(https://cdn.filestackcontent.com/resize=width:24,height:24/enCa5BTVeIsLHcgkdpAc); font-size: 0.625rem; height: 1.5rem; width: 1.5rem;"
        >
          <!---->
        </div>
         
        <div
          class="user-avatar__details"
        >
          <div
            class="user-avatar__name"
          >
            Xavier Oaxaca
          </div>
           
          <div
            class="user-avatar__title"
          >
            HR Manager
          </div>
        </div>
      </div>
    </div>
    <div
      class="my-4"
    >
      <h2
        class="type-overline type-overline-small"
      >
        medium
      </h2>
       
      <div
        class="user-avatar user-avatar--medium user-avatar--horizontal"
      >
        <div
          class="avatar-image avatar-image--rounded"
          style="background-image: url(https://cdn.filestackcontent.com/resize=width:32,height:32/enCa5BTVeIsLHcgkdpAc); font-size: 0.875rem; height: 2rem; width: 2rem;"
        >
          <!---->
        </div>
         
        <div
          class="user-avatar__details"
        >
          <div
            class="user-avatar__name"
          >
            Xavier Oaxaca
          </div>
           
          <div
            class="user-avatar__title"
          >
            HR Manager
          </div>
        </div>
      </div>
    </div>
    <div
      class="my-4"
    >
      <h2
        class="type-overline type-overline-small"
      >
        large
      </h2>
       
      <div
        class="user-avatar user-avatar--large user-avatar--horizontal"
      >
        <div
          class="avatar-image avatar-image--rounded"
          style="background-image: url(https://cdn.filestackcontent.com/resize=width:40,height:40/enCa5BTVeIsLHcgkdpAc); font-size: 1.0625rem; height: 2.5rem; width: 2.5rem;"
        >
          <!---->
        </div>
         
        <div
          class="user-avatar__details"
        >
          <div
            class="user-avatar__name"
          >
            Xavier Oaxaca
          </div>
           
          <div
            class="user-avatar__title"
          >
            HR Manager
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mr-12"
  >
    <h1
      class="type-overline type-overline-large"
    >
      
        vertical
      
    </h1>
     
    <div
      class="my-4"
    >
      <h2
        class="type-overline type-overline-small"
      >
        tiny
      </h2>
       
      <div
        class="user-avatar user-avatar--tiny user-avatar--vertical"
      >
        <div
          class="avatar-image avatar-image--rounded"
          style="background-image: url(https://cdn.filestackcontent.com/resize=width:48,height:48/enCa5BTVeIsLHcgkdpAc); font-size: 1.3125rem; height: 3rem; width: 3rem;"
        >
          <!---->
        </div>
         
        <div
          class="user-avatar__details"
        >
          <div
            class="user-avatar__name"
          >
            Xavier Oaxaca
          </div>
           
          <div
            class="user-avatar__title"
          >
            HR Manager
          </div>
        </div>
      </div>
    </div>
    <div
      class="my-4"
    >
      <h2
        class="type-overline type-overline-small"
      >
        small
      </h2>
       
      <div
        class="user-avatar user-avatar--small user-avatar--vertical"
      >
        <div
          class="avatar-image avatar-image--rounded"
          style="background-image: url(https://cdn.filestackcontent.com/resize=width:48,height:48/enCa5BTVeIsLHcgkdpAc); font-size: 1.3125rem; height: 3rem; width: 3rem;"
        >
          <!---->
        </div>
         
        <div
          class="user-avatar__details"
        >
          <div
            class="user-avatar__name"
          >
            Xavier Oaxaca
          </div>
           
          <div
            class="user-avatar__title"
          >
            HR Manager
          </div>
        </div>
      </div>
    </div>
    <div
      class="my-4"
    >
      <h2
        class="type-overline type-overline-small"
      >
        medium
      </h2>
       
      <div
        class="user-avatar user-avatar--medium user-avatar--vertical"
      >
        <div
          class="avatar-image avatar-image--rounded"
          style="background-image: url(https://cdn.filestackcontent.com/resize=width:64,height:64/enCa5BTVeIsLHcgkdpAc); font-size: 1.75rem; height: 4rem; width: 4rem;"
        >
          <!---->
        </div>
         
        <div
          class="user-avatar__details"
        >
          <div
            class="user-avatar__name"
          >
            Xavier Oaxaca
          </div>
           
          <div
            class="user-avatar__title"
          >
            HR Manager
          </div>
        </div>
      </div>
    </div>
    <div
      class="my-4"
    >
      <h2
        class="type-overline type-overline-small"
      >
        large
      </h2>
       
      <div
        class="user-avatar user-avatar--large user-avatar--vertical"
      >
        <div
          class="avatar-image avatar-image--rounded"
          style="background-image: url(https://cdn.filestackcontent.com/resize=width:104,height:104/enCa5BTVeIsLHcgkdpAc); font-size: 2.8125rem; height: 6.5rem; width: 6.5rem;"
        >
          <!---->
        </div>
         
        <div
          class="user-avatar__details"
        >
          <div
            class="user-avatar__name"
          >
            Xavier Oaxaca
          </div>
           
          <div
            class="user-avatar__title"
          >
            HR Manager
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
